### PR TITLE
create zed command

### PR DESCRIPTION
**Pull request by mccanne**
This commit is the initial stab at the zed command.
In this first commit, the zq, ast, zst, and zapi
commands have been moved under zed as "zed q",
"zed ast", "zed zst", and "zed api".  Commands for
"zq", "zast", and "zapi" have been left in place
as stand-alone shortcuts for their zed counterparts.
The charm package made this refactoring particularly
easy.

### CODE REVIEW

**Review by mattnibs — APPROVED** [review]
😎

File: cmd/zed/q/command.go [hunk 1]
```diff
@@ -47,34 +51,31 @@ match input types.  If multiple files are concatenated into a stream and
...
+If the first argument is a path to a valid file rather than a zed query,
+then the zed query is assumed to be "*", i.e., match and output all
+of the input.  If the first argument is both a valid zed query
+and an existing file, then the file overrides.
 
-The Z query text may include files using -I, which is particularly
+The zed query text may include files using -I, which is particularly
 convenient when a large, complex query spans multiple lines.  In this case,
-these Z files are concatenated together along with the command-line Z text
+these Zed files are concatenated together along with the command-line zed query text
```
Comment (nwt) [inline cmd/zed/q/command.go:67]:
```suggestion
these zed files are concatenated together along with the command-line zed query text
```

**Review by nwt — APPROVED** [review]

File: pkg/charm/help.go [hunk 2]
```diff
@@ -10,6 +11,8 @@ import (
 	"github.com/kr/text"
 )
 
+var NeedHelp = errors.New("help")
```
[the remark this answers is not in the document]
Comment (mccanne) [reply pkg/charm/help.go:14]: It's not an error in the same way context.Canceled isn't an error.

[the remark this answers is not in the document]
Comment (nwt) [reply pkg/charm/help.go:14]: Fair enough.

**Pull request merged**